esp-hal/src/dma/mod.rs

```diff
@@ -1585,13 +1593,46 @@ impl RxCircularState {
     }
 }
 
+#[doc(hidden)]
+pub trait DmaRxChannel:
+    RxRegisterAccess + InterruptAccess<DmaRxInterrupt> + Peripheral<P = Self>
+{
+}
+
+#[doc(hidden)]
+pub trait DmaTxChannel:
+    TxRegisterAccess + InterruptAccess<DmaTxInterrupt> + Peripheral<P = Self>
+{
+}
+
 /// A description of a DMA Channel.
-pub trait DmaChannel: crate::private::Sealed + Sized {
+pub trait DmaChannel: Peripheral<P = Self> {
     /// A description of the RX half of a DMA Channel.
-    type Rx: RxRegisterAccess + InterruptAccess<DmaRxInterrupt>;
+    type Rx: DmaRxChannel;
 
     /// A description of the TX half of a DMA Channel.
-    type Tx: TxRegisterAccess + InterruptAccess<DmaTxInterrupt>;
+    type Tx: DmaTxChannel;
+
+    /// Sets the priority of the DMA channel.
+    #[cfg(gdma)]
+    fn set_priority(&self, priority: DmaPriority);
+
+    /// Splits the DMA channel into its RX and TX halves.
+    #[cfg(any(esp32c6, esp32h2, esp32s3))] // TODO relax this to allow splitting on all chips
+    fn split(self) -> (Self::Rx, Self::Tx) {
```

Comment:
Something is missing still, so this isn't useable. The upcoming PRs should fix that.